src/bsdutils/meson.build

```diff
@@ -1,18 +1,9 @@
-
-# Required on Alpine Linux (and others?)
-# TODO: The "required" kwarg should be "true" but only for systems
-# that require it, such as Alpine Linux
-#
-# This doesn't work because meson.get_external_property() returns 'glibc'
-#
-#dep_musl_fts = []
-#libc_type = meson.get_external_property('libc', 'glibc').to_lower()
-#if libc_type == 'musl' or libc_type == 'musl-libc'
-#  dep_musl_fts = dependency('musl-fts', required: true)
-#endif
-#message(libc_type)
-
-dep_musl_fts = dependency('musl-fts', required: false)
+dep_musl_fts = []
+if cc.has_header_symbol('fts.h', 'fts_open')
+  dep_musl_fts = cc.find_library('fts', required: false)
```

Comment:
I would say that you still want to use the dependency() for it.

As it is, you're checking for a library even when fts.h is provided by libc, which isn't really what I suggested. Instead, you can do one of a couple things:
- cc.links() using a custom code snippet that implements a toy use of this function
- check if the header exists and the binary symbol is provided without any libraries

If those fail, then try to find an external dependency.

Reply:
> I would say that you still want to use the dependency() for it.
> 
> As it is, you're checking for a library even when fts.h is provided by libc, which isn't really what I suggested.

I tried what you suggested, but the unusual thing (at least to me it seems unusual) is that on Alpine, when has_header() is successful, I still have to use `dependency()` or `find_library()`, otherwise the linking fails (with undefined reference errors).

And on Linux, FreeBSD, and macos, has_header() succeeds, and trying to find an 'fts' or 'musl-fts' dependency fails (because it doesn't exist and isn't needed).

>Instead, you can do one of a couple things:
> 
>     * cc.links() using a custom code snippet that implements a toy use of this function
> 
>     * check if the header exists and the binary symbol is provided without any libraries
> 
> 
> If those fail, then try to find an external dependency.

I'll have a look at those options; I'm not super-happy about how it is now either ;) Thanks again @eli-schwartz

Reply:
> Instead, you can do one of a couple things:
> - cc.links() using a custom code snippet that implements a toy use of this function
> - check if the header exists and the binary symbol is provided without any libraries

Just want to reiterate that option 2 is valid, and looks like this:
```meson
cc.has_function('fts_open', prefix: '#include <fts.h>')
```